qadence/transpile/reverse.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from functools import singledispatch
+from typing import overload
+
+from qadence import QuantumCircuit
+from qadence.blocks import AbstractBlock, ChainBlock, chain, tag
+
+
+@overload
+def reverse(circuit: QuantumCircuit) -> QuantumCircuit:
+    ...
+
+
+@overload
+def reverse(block: AbstractBlock) -> AbstractBlock:
+    ...
+
+
+@singledispatch
+def reverse(x: QuantumCircuit | AbstractBlock) -> QuantumCircuit | AbstractBlock:
+    """Reverses a QuantumCircuit or AbstractBlock."""
+    raise NotImplementedError(f"Unable to invert endianness of object {type(x)}.")
+
+
+@reverse.register(AbstractBlock)  # type: ignore[attr-defined]
+def _(block: AbstractBlock) -> AbstractBlock:
+    """Reverses a block if its a ChainBlock."""
+    if isinstance(block, ChainBlock):
+        blk = chain(*(reverse(b) for b in reversed(block.blocks)))
+        return tag(blk, block.tag) if block.tag is not None else blk
+    else:
+        return block
+
+
+@reverse.register(QuantumCircuit)  # type: ignore[attr-defined]
+def _(circuit: QuantumCircuit) -> QuantumCircuit:
```

Comment:
we have a `reverse_program` in `transpile/circuit.py` (which is not implemented). Should we call this function `reverse_program` and remove the other not-implemented one? Would also argue for a name change because `reversed` already exists in python